WORKSPACE

```diff
@@ -121,12 +139,34 @@ http_archive(
 # gRPC.
 http_archive(
     name = "com_github_grpc_grpc",
+    patch_args = ["-p1"],
+    patches = [
+        # To maintain compatibility with python 3.10 and greater, we need to patch
+        # in the following grpc change:
+        # https://github.com/grpc/grpc/commit/dbe73c9004e483d24168c220cd589fe1824e72bc
+        #
+        # To reproduce the patch:
+        # ```
+        # $ git clone https://github.com/grpc/grpc.git
+        # $ cd grpc
+        # $ git checkout tags/v1.27.0-pre1 -b my-patch
+        # $ git cherry-pick dbe73c9004e483d24168c220cd589fe1824e72bc
+        # $ git diff HEAD~1 > grpc.patch
+        # ```
+        #
+        # Note that we choose tags/v1.27.0-pre1 as the base since the version in
+        # the archive is 1.27.0-dev. It's not necessarily an exact match but is
```

Comment:
For future reference, the prefix of the archive that we strip (`grpc-b54a5b338637f92bfcf4b0bc05e0f57a5fd8fadd`) includes the commit hash, so we can always check out specifically that commit if you want an exact match.

Reply:
Thanks. I'm adjusting the instructions in https://github.com/tensorflow/tensorboard/pull/5795.